### PR TITLE
doc: Update 'use_git' example

### PR DESCRIPTION
**Pull request by tommygarvin**
## Changes

- Fix: Update `use_git` example where `boolean` should be `bool` 
- Fix: Update `use_git` example to remove `choices` field
- Fix: Update Supported Types "text" section so CTRL-D is rendered
- Style: Replace static text with a reference to 'supported types' section
- Style: Adjust whitespace

### use_git example

If you attempt to run the `use_git` example as it is currently you'll receive
> Error: invalid type for variable `use_git`: `boolean` possible values are `bool` and `string`

And, after you change it from `boolean` to `bool` you'll receive
> Error: bool type does not support `choices` field

### CTRL-D

The rendered page does not show CTRL-D
> "text": Represents a multiline string value. (terminated by hitting )

### CODE REVIEW

**Comment by tommygarvin** [comment]
Any feedback @sassman & @joshrotenberg ?

File: guide/src/templates/template_defined_placeholders.md [hunk 1]
```diff
@@ -40,19 +40,18 @@ Consider the following `cargo-generate.toml` file:
 ```toml
 [placeholders]
 project_name = { prompt = "Enter project name", default = "my_project", type = "string" }
-use_git = { prompt = "Initialize Git repository?", choices = ["Yes", "No"], default = "Yes", type = "boolean" }
+use_git = { prompt = "Initialize Git repository?", default = true, type = "bool" }
```
Comment (sassman) [inline guide/src/templates/template_defined_placeholders.md:43]: logically you're making sense here. However, the intent was to showcase the `choices`. 
So the better fix would be `type = "string"`

But maybe we keep this, and add a different example for `choices`.

File: guide/src/templates/template_defined_placeholders.md [hunk 2]
```diff
@@ -15,23 +15,23 @@ placeholder_name = { prompt = "Enter your name", choices = ["Alice", "Bob"], def
...
 ## Supported Types
 
 `cargo-generate` supports the following placeholder value types:
 
 - `"string"`: Represents a string value.
-- `"text"`: Represents a multiline string value. (terminated by hitting <CTRL-D>)
+- `"text"`: Represents a multiline string value. (terminated by hitting CTRL-D)
 - `"editor"`: Represents a multiline string value, collected from the user by a real terminal editor.
-- `"boolean"`: Represents a boolean value (`true` or `false`).
+- `"bool"`: Represents a boolean value (`true` or `false`).
```
Comment (sassman) [inline guide/src/templates/template_defined_placeholders.md:34]: good catch!

**Comment by sassman** [comment]
Hey @tommygarvin,

thanks for your efforts. Do you think you can have a look at the merge conflicts and get them sorted? 
Then we could proceed landing this one.

**Comment by sassman** [comment]
@tommygarvin could you pls have a look at the merge conflicts, then I would like to merge it.

File: guide/src/templates/template_defined_placeholders.md (hunk 1 above)
Comment (tommygarvin) [reply guide/src/templates/template_defined_placeholders.md:44]: Ah, makes sense. I added a new example showcasing the choices feature:  https://github.com/cargo-generate/cargo-generate/pull/1191/commits/887df5f64d3a8e8f84f3ac5028b115773604fb70

**Comment by tommygarvin** [comment]
Should be all clear. Thanks, @sassman !

**Review by sassman — APPROVED** [review]

**Comment by sassman** [comment]
Thanks a lot @tommygarvin for your contribution. 🫶

**Pull request merged**